File: projekat-eo/src/main/java/com/spring/projekateo/model/Document.java
package com.spring.projekateo.model;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.Table;

@Entity
@Table(name = "document")
public class Document {

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	@Column(name="document_id",unique = true, nullable = false)
	private Integer id;
	
	@Column(name = "title", nullable=false)
	private String title;
	
	@Column(name="url", nullable=false)
	private String url;

	@ManyToOne
	@JoinColumn(name = "student_id", referencedColumnName = "student_id", nullable = false)
	private Student student;
	
	@ManyToOne
	@JoinColumn(name = "type_id", referencedColumnName = "document_type_id", nullable = false)
	private DocumentType type;
	
	@Column(name = "active", nullable = false)
	private boolean active = true;
	
	public Document() {
		
	}

	public Integer getId() {
		return id;
	}

	public void setId(Integer id) {
		this.id = id;
	}

	public String getTitle() {
		return title;
	}

	public void setTitle(String title) {
		this.title = title;
	}

	public String getUrl() {
		return url;
	}

	public void setUrl(String url) {
		this.url = url;
	}

	public Student getStudent() {
		return student;
	}

	public void setStudent(Student student) {
		this.student = student;
	}

	public DocumentType getType() {
		return type;
	}

	public void setType(DocumentType type) {
		this.type = type;
	}

	public boolean isActive() {
		return active;
	}

	public void setActive(boolean active) {
		this.active = active;
	}
	
}

File: projekat-eo-angular/angularApp/src/app/model/document.ts
import { DocumentType } from "./document-type";
import { Student } from "./student";

export class Document implements DocumentInterface{
	public id: number;
	public title: string;
	public url: string;
    public type: DocumentType;
	public student: Student;
		
	constructor(documentCfg: DocumentInterface)
	{	
		this.id = documentCfg.id;
		this.title = documentCfg.title;
		this.url = documentCfg.url;
        this.type = documentCfg.type;
		this.student = documentCfg.student;
	}
}

interface DocumentInterface {
	id?: number;
	title: string;
	url: string;
    type: DocumentType;
	student: Student;
}